File: Scores/Interfaces/IStudentMapper.cs
using Scores.Models;

namespace Scores.Interfaces
{
    public interface IStudentMapper
    {
        Task<List<StudentSubject>> MapToStudentSubjectsAsync(List<StudentScoreRow> rows);
        Task<List<StudentSubject>> MapAndSortScoresAsync(List<StudentScoreRow> rows);
    }
}

File: scores-php/src/Interfaces/IStudentMapper.php
<?php

namespace Scores\Interfaces;

interface IStudentMapper
{
    public function mapToStudentSubjects(array $rows): array;
    
    public function mapAndSortScores(array $rows): array;
    
    public function mapAndSortScoresFromIterable(iterable $rows): array;
}

